alica_engine/src/engine/modelmanagement/ModelManager.cpp

```diff
@@ -28,27 +29,41 @@ namespace alica
 
 ModelManager::ModelManager(PlanRepository& planRepository, AlicaEngine* ae, const std::string& domainConfigFolder)
         : _planRepository(planRepository)
-        , _ae(ae)
+        , _configChangeListener(ae->getConfigChangeListener()) // tmp only for compilation
         , domainConfigFolder(domainConfigFolder)
 {
     auto reloadFunctionPtr = std::bind(&ModelManager::reload, this, std::placeholders::_1);
-    _ae->subscribe(reloadFunctionPtr);
-    reload(_ae->getConfig());
+    _configChangeListener.subscribe(reloadFunctionPtr);
+    reload(_configChangeListener.getConfig());
+    Factory::setModelManager(this);
+}
+
+ModelManager::ModelManager(PlanRepository& planRepository, ConfigChangeListener& configChangeListener, const std::string& domainConfigFolder)
+        : _planRepository(planRepository)
+        , _configChangeListener(configChangeListener)
+        , domainConfigFolder(domainConfigFolder)
+{
+    auto reloadFunctionPtr = std::bind(&ModelManager::reload, this, std::placeholders::_1);
+    _configChangeListener.subscribe(reloadFunctionPtr);
+    reload(_configChangeListener.getConfig());
     Factory::setModelManager(this);
 }
 
 void ModelManager::reload(const YAML::Node& config)
 {
+    YAML::Node& originalConfig = _configChangeListener.getConfig();
+    originalConfig = config;
```

Comment:
What's the point of doing this? Also incoming argument of config could be further passed to getBasePath.

Reply:
Just something that is forgotten during a test